controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -202,32 +202,27 @@ public void initializeStreamWriters(final ClientFactory clientFactory,
                 .thenCompose(retentionSet -> {
                     StreamCutRecord latestCut = retentionSet.stream()
                             .max(Comparator.comparingLong(StreamCutRecord::getRecordingTime)).orElse(null);
-                    return checkGenerateStreamCut(scope, stream, context, policy, latestCut, recordingTime)
-                            .thenCompose(x -> truncate(scope, stream, policy, context, retentionSet, recordingTime));
+                    return checkGenerateStreamCut(scope, stream, context, latestCut, recordingTime)
```

Comment:
I was not sure where to leave this comment, but I'm wondering if we should we look into having metrics for retention so that we can track progress of stream retention via UI. What do you think?

Reply:
separate issue?

Reply:
I won't block the PR on this point, but it is best that we start making metrics part an integral part of feature PRs.